src/robotFactory.js

```diff
@@ -1,13 +1,75 @@
-'use strict';
-
-class BaseRobot {}
-
-class FlyingRobot {}
-
-class DeliveryDrone {}
-
-module.exports = {
-  BaseRobot,
-  FlyingRobot,
-  DeliveryDrone,
-};
+'use strict';
+
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;

```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    }
```
You can combine all coords like this